src/internal/healthchecks/healthchecks.go

```diff
@@ -32,9 +38,97 @@ func Run(ctx context.Context, watcher watcher.StatusWatcher, healthChecks []v1al
 		}
 		objs = append(objs, obj)
 	}
-	err := pkgkubernetes.WaitForReady(ctx, watcher, objs)
+	err := WaitForReady(ctx, watcher, objs)
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+// WaitForReadyRuntime waits for all of the objects to reach a ready state.
+func WaitForReadyRuntime(ctx context.Context, sw watcher.StatusWatcher, robjs []runtime.Object) error {
+	objs := []object.ObjMetadata{}
+	for _, robj := range robjs {
+		obj, err := object.RuntimeToObjMeta(robj)
+		if err != nil {
+			return err
+		}
+		objs = append(objs, obj)
+	}
+	return WaitForReady(ctx, sw, objs)
+}
+
+// WaitForReady waits for all of the objects to reach a ready state.
+func WaitForReady(ctx context.Context, sw watcher.StatusWatcher, objs []object.ObjMetadata) error {
+	cancelCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	eventCh := sw.Watch(cancelCtx, objs, watcher.Options{})
+	statusCollector := collector.NewResourceStatusCollector(objs)
+	done := statusCollector.ListenWithObserver(eventCh, collector.ObserverFunc(
+		func(statusCollector *collector.ResourceStatusCollector, _ event.Event) {
+			rss := []*event.ResourceStatus{}
+			for _, rs := range statusCollector.ResourceStatuses {
+				if rs == nil {
+					continue
+				}
+				rss = append(rss, rs)
+			}
+			desired := status.CurrentStatus
+			if aggregator.AggregateStatus(rss, desired) == desired {
+				cancel()
+				return
+			}
+		}),
+	)
+	<-done
+
+	for _, id := range objs {
+		rs := statusCollector.ResourceStatuses[id]
+		switch rs.Status {
+		case status.CurrentStatus:
+			message.Debugf("%s: %s ready", rs.Identifier.Name, strings.ToLower(rs.Identifier.GroupKind.Kind))
+		case status.NotFoundStatus:
+			message.Warnf("%s: %s not found", rs.Identifier.Name, strings.ToLower(rs.Identifier.GroupKind.Kind))
+		default:
+			message.Warnf("%s: %s not ready", rs.Identifier.Name, strings.ToLower(rs.Identifier.GroupKind.Kind))
```

Comment:
I think all of these should be Debug to not pollute the output even more.

Reply:
I disagree, when either of these situations are encountered a generic error will be returned and the deployment will fail. In that case I think we should communicate what went wrong.

Reply:
Maybe instead of these logs we should craft a custom error message with the not found / not ready resources?

Reply:
Changed to return the not found and not ready resources within the error